NineChronicles.Headless/GraphTypes/StateQuery.cs

```diff
@@ -40,18 +56,27 @@ public StateQuery()
                 resolve: context =>
                 {
                     var address = context.GetArgument<Address>("avatarAddress");
-                    try
-                    {
-                        return new AvatarStateType.AvatarStateContext(
-                            context.Source.AccountState.GetAvatarState(address),
-                            context.Source.AccountState,
-                            context.Source.BlockIndex);
-                    }
-                    catch (InvalidAddressException)
+                    return GetAvatarState(context.Source, address)
+                        ?? throw new InvalidOperationException($"The state {address} doesn't exists");
+                });
+            Field<NonNullGraphType<ListGraphType<AvatarStateType>>>(
+                name: "avatars",
+                description: "Avatar states having some order as addresses",
+                arguments: new QueryArguments(
+                    new QueryArgument<NonNullGraphType<ListGraphType<AddressType>>>
```

Comment:
If you don't want to receive `[null]`, it should be:

```suggestion
                    new QueryArgument<NonNullGraphType<ListGraphType<NonNullGraphType<AddressType>>>>
```